lib/hackmode/realmode.js

```diff
@@ -284,143 +274,84 @@ class N3hRealMode extends N3hMode {
-        //         content: metaContent
-        //       })).toString('base64')
-        //     })
-        //     log.t(this.me() + ' publish Meta', metaId)
-        //     this._bookkeepAddress(this._publishedMetaBook, data.dnaAddress, metaId)
-        //   }
-        //   return
-
-        // case 'fetchEntry':
-        //   // Note: data is a FetchEntryData
-        //   // log.t(this.me() + ' fetchEntry', result)
-        //   if (!this._hasTrackOrFail(data.requesterAgentId, data.dnaAddress, data._id)) {
-        //     return
-        //   }
-        //
-        //   ipcMsg = {
-        //     method: 'failureResult',
-        //     _id: data._id,
-        //     dnaAddress: data.dnaAddress,
-        //     toAgentId: data.requesterAgentId,
-        //     resultInfo: 'Entry not found:' + data.address
-        //   }
-        //
-        //   // Ask my dna DHT for the data
-        //   // TODO: might need to do dht.post(DhtEvent.fetchData) instead for non-fullsync DHT
-        //   result = await this._getDhtRef(data.dnaAddress).fetchData(data.address)
-        //   // log.d(this.me() + ' fetchData', result)
-        //   if (result && Array.isArray(result)) {
-        //     for (let item of result) {
-        //       const res = JSON.parse(Buffer.from(item, 'base64').toString())
-        //       if (res && res.type === 'dhtEntry') {
-        //         ipcMsg = {
-        //           method: 'fetchEntryResult',
-        //           _id: data._id,
-        //           dnaAddress: data.dnaAddress,
-        //           requesterAgentId: data.requesterAgentId,
-        //           providerAgentId: res.providerAgentId,
-        //           address: res.address,
-        //           content: res.content
-        //         }
-        //         break
-        //       }
-        //     }
-        //   }
-        //   log.d(this.me() + ' fetchData', result)
-        //   this._ipcSend('json', ipcMsg)
-        //   return
-
-      // #fullsync hack: send request back to self (ipc)
-      case 'fetchEntry':
-        // Note: data is a FetchEntryData
+      case 'queryEntry':
+        // Note: data is a QueryEntryData
         if (!this._hasTrackOrFail(data.requesterAgentId, data.dnaAddress, data._id)) {
           return
         }
-        //  since we're fully connected, just redirect this back to itself for now...
-        data.method = 'handleFetchEntry'
+        // #fullsync hack: send request back to self (ipc)
+        data.method = 'handleQueryEntry'
+        this._ipcSend('json', data)
+        return
+      case 'handleQueryEntryResult':
+        // Note: data is a QueryEntryResultData
+        if (!this._hasTrackOrFail(data.responderAgentId, data.dnaAddress, data._id)) {
+          return
+        }
+        // #fullsync hack: send response back to self (ipc)
+        data.method = 'queryEntryResult'
         this._ipcSend('json', data)
         return
       case 'handleFetchEntryResult':
```

Comment:
Should this be handleQueryEntryResult now?

Reply:
???
`handleQueryEntryResult` is handled just above.
`handleFetchEntryResult` is for resolving a data request initiated by gossip